source/codegen/templates/service.h.mako

```diff
@@ -56,8 +57,13 @@ private:
 % endif
 % if 'custom_types' in locals():
 %   for custom_type in custom_types:
+	% if custom_type["name"] in output_custom_types:
   void Copy(const ${custom_type["name"]}& input, ${namespace_prefix}${custom_type["grpc_name"]}* output);
   void Copy(const std::vector<${custom_type["name"]}>& input, google::protobuf::RepeatedPtrField<${namespace_prefix}${custom_type["grpc_name"]}>* output);
+	% endif
+	% if custom_type["name"] in input_custom_types:
+  ${custom_type["name"]} GetStructFromGrpcType(const ${namespace_prefix}${custom_type["grpc_name"]}& input);
```

Comment:
We should be consistent with the other `Copy` methods. Just above we define a copy from struct to grpc type, and another from vector of structs to `RepeatedPtrField`. I'd propose renaming your `GetStructFromGrpcType` to `ConvertMessage` and add a new `Copy` as follows:
```cpp
CustomStruct ConvertMessage(const GrpcType& input);
void Copy(const RepeatedPtrField<GrpcType>& input, std::vector<CustomStruct>& output);
```
Then you would move the call to `std::transform` from the service handler into the new `Copy` function.

Reply:
We have been following pattern of using inline `transform` calls for copying input array params, and `Copy` functions for copying output array params for `ViSession` and `ViBoolean` types as well.

I like the idea of renaming the function from `GetStructFromGrpcType` to `ConvertMessage` though.